upp/stages/hist.py

```diff
@@ -104,7 +104,13 @@ def main(config=None):
         log.info(f"Estimating PDF for {c}")
         c.setup_reader(config.batch_size)
         cuts_no_split = c.cuts.ignore(["eventNumber"])
-        c.check_num_jets(config.num_jets_estimate, cuts=cuts_no_split, silent=True)
+        try:
+            c.check_num_jets(config.num_jets_estimate, cuts=cuts_no_split, silent=True)
+        except ValueError as e:
+            log.warning(e)
+            log.warning(
+                "Ran out of jets when estimating PDFs, listed amount is total available jets"
```

Comment:
Looks good thanks! can we make this warning a bit more clear? By "listed amount" do you whatever is printed in `log.warning(e)`? How many jets are actually used to calculate the histogram in the end? 

Perhaps something like
```
f"{config.num_jets_estimate:,} jets not available, using maximum available jets ({max_available:,}) to calculate PDF"
```

Reply:
Yes by 'listed amount' I'm referring to the total number of jets available, so the number you get when the exception says "but only XXX are estimated to be in sample".  This can then be used to decide total number of each jet flavour desired for further preprocessing steps.

For calculating the histograms, we still use 'num_jets_estimate', but I think this number is the total number of jets loaded, not number of jets in that sample, so we aren't necessarily using (in fact, probably are not using) all jets of a given sample to calculate the hist.

Reply:
I'm probably forgetting how this works, but if `num_jets_estimate` jets aren't available, won't L114 still error due to not having enough jets?

Reply:
Nope. L114 -> [components.get_jets](https://github.com/umami-hep/umami-preprocessing/blob/main/upp/classes/components.py#L51) -> [reader.load](https://github.com/umami-hep/atlas-ftag-tools/blob/main/ftag/hdf5/h5reader.py#L285) -> [reader.stream](https://github.com/umami-hep/atlas-ftag-tools/blob/main/ftag/hdf5/h5reader.py#L215) [singlereader.stream](https://github.com/umami-hep/atlas-ftag-tools/blob/main/ftag/hdf5/h5reader.py#L70).
Important to note, as far as i can tell, is that we request the number of total jets before the cuts are applied. So if 'num_jets_estimate=2,000,000' we load 2,000,000 jets, apply flavour cuts, and then use the remaining jets to calculate the hists. I think this means that the line I'm putting an exception around exists purely to let the user know how many jets of that component exists.